### PR TITLE
P20-1055: Rename user CAP columns from `state` to `status`

### PR DESCRIPTION
**Pull request by artem-vavilov**
Renamed user columns `cap_state` & `cap_state_date` to `cap_status` & `cap_status_date`, along with the corresponding index to not confuse them with `U.S. state`.

#### `CAP` - Child Account Policy
https://github.com/code-dot-org/code-dot-org/blob/180cf779efa5c7bc0918ca17ca0a3dd0e4933fb6/dashboard/config/initializers/inflections.rb#L22

## Links
- JIRA task: [P20-1055](https://codedotorg.atlassian.net/browse/P20-1055)
- Migration PR: https://github.com/code-dot-org/code-dot-org/pull/60083
- Phase 1 PR: https://github.com/code-dot-org/code-dot-org/pull/59962
- Phase 2 PR: https://github.com/code-dot-org/code-dot-org/pull/59963

### CODE REVIEW

**Review by mgc1194 — APPROVED** [review]
The changes look good. I'd like at least an extra approval to make sure there are no consequences I'm unaware of.
I see the following PRs have tests and further updates to the usage of these columns.

**Review by nataliazm99 — APPROVED** [review]
The column names look updated, I don't foresee issues given that these fields are not in use yet. Please check wtih infra for any other downstream effects.

**Review by daynew — APPROVED** [review]

File: dashboard/db/migrate/20240806035905_rename_cap_columns_in_users.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class RenameCAPColumnsInUsers < ActiveRecord::Migration[6.1]
+  def change
+    rename_column :users, :cap_state, :cap_status
+    rename_column :users, :cap_state_date, :cap_status_date
+  end
+end
```
Comment (cat5inthecradle) [inline dashboard/db/migrate/20240806035905_rename_cap_columns_in_users.rb:6]: @code-dot-org/infrastructure Do we expect this to have a significant performance impact, or be too slow for a regular migration during the DTP?

**Comment by cat5inthecradle** [comment]
How have you tested this migration? This might go smooth and quick, but I'm concerned that because these are used in an index on a very large table, it might be problematic.

**Comment by cat5inthecradle** [comment]
sorry, that's the second time this week I've closed a PR by accident :(

File: dashboard/db/migrate/20240806035905_rename_cap_columns_in_users.rb (hunk 1 above)
Comment (Hamms) [reply dashboard/db/migrate/20240806035905_rename_cap_columns_in_users.rb:6]: This should be just fine! Definitely important to double-check, but we get column and index rename operations for free on our current version of MySQL. Fortunately, I already had a clone of the production db lying around I could use to verify:

```
ubuntu@adhoc-cap-data-migration-testing:~/adhoc/dashboard$ RAILS_ENV=adhoc RACK_ENV=adhoc bundle exec rake db:migrate
GetSecretValue: adhoc/cdo/recaptcha_site_key
GetSecretValue: adhoc/cdo/recaptcha_secret_key
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
== 20240806035905 RenameCAPColumnsInUsers: migrating ==========================
-- rename_column(:users, :cap_state, :cap_status)
   -> 0.8567s
-- rename_column(:users, :cap_state_date, :cap_status_date)
   -> 0.7402s
== 20240806035905 RenameCAPColumnsInUsers: migrated (1.5972s) =================

WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
```

**Review by Hamms — APPROVED** [review]

**Review by cat5inthecradle — APPROVED** [review]

**Pull request merged**